include/seqan/index/find2_index_approx.h

```diff
@@ -295,16 +331,89 @@ inline void _optimalSearchSchemeComputeFixedBlocklength(std::array<OptimalSearch
         blocklengths.push_back(blocklength + (i < rest));
 
     _optimalSearchSchemeSetBlockLength(ss, blocklengths);
+    _optimalSearchSchemeComputeBlockBorders(ss);
     _optimalSearchSchemeInit(ss);
 }
 
-template <typename TDelegate,
+//NOTE (svnbngk) Compare potential occurrences directly to genome if the range on the index is small enough.
+template <typename TDelegateD,
+          typename TText, typename TIndex, typename TIndexSpec,
+          typename TNeedle,
+          size_t nbrBlocks,
+          typename TDir>
+inline void inTextVerification(TDelegateD & delegateDirect,
+                               Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
+                               TNeedle const & needle,
+                               uint32_t const needleLeftPos,
+                               uint32_t const needleRightPos,
+                               uint8_t const errors,
+                               OptimalSearch<nbrBlocks> const & s,
+                               uint8_t const blockIndex,
+                               TDir const & /**/)
+{
+    auto const & genome = indexText(*iter.fwdIter.index);
+
+    //cut of blockStarts and Ends that where already checked by the search
+    std::vector<uint32_t> blockStarts(s.pi.size() - blockIndex);
```

Comment:
If you replace this with an array my following comment is not needed here but might be good to know in general: Instead of creating the vector and then copying the values over you can directly create an vector by giving begin end end iterator of another range (https://en.cppreference.com/w/cpp/container/vector/vector). :)